marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/ExecEndpointImpl.java

```diff
@@ -166,32 +171,78 @@ private boolean processOutput(CallContextImpl callContext){
                         callContext.withEndpointState(output);
                     }
                     incrementCallCount();
-                } catch(Throwable throwable) {
+                    return output;
+                } catch(Throwable catchedThrowable) {
                     // TODO: logging
-                    throw new RuntimeException("error while calling "+getEndpoint().getEndpointPath(), throwable);
+                    throwable = catchedThrowable;
                 }
-// TODO -- retry with new session if times out
+                // TODO -- retry with new session if times out
```

Comment:
We can remove the comment.  I believe there's no error.  The appserver just sees it as a request to create a new session if the session times out.

Reply:
Got it. Removed it.